app/workers/rapid_ready_for_decision/rrd_processor.rb

```diff
@@ -10,34 +12,53 @@ def initialize(form526_submission)
     end
 
     def run
-      assess_data
+      use_vro = Flipper.enabled?(:rrd_call_vro_service)
+      use_vro ? assess_data_with_vro : assess_data
+
       unless @claim_context.sufficient_evidence
         return @form526_submission.save_metadata(offramp_reason: 'insufficient_data')
       end
 
       add_medical_stats
 
-      pdf = generate_pdf
+      pdf_body = (use_vro ? generate_pdf_body_with_vro : generate_pdf.render)
       @claim_context.add_metadata(pdf_created: true)
-      upload_pdf(pdf)
+      upload_pdf(pdf_body)
 
       set_special_issue if Flipper.enabled?(:rrd_add_special_issue) && release_pdf?
 
       @claim_context.save_metadata
     end
 
     # Populates @claim_context.assessed_data and sets claim_context.sufficient_evidence
-    # Return nil to discontinue processing (i.e., doesn't generate pdf or set special issue)
     def assess_data
       raise "Method `assess_data` should be overriden by the subclass #{self.class}"
     end
 
+    # Returns whether or not @claim_context.assessed_data contains sufficient evidence
+    def sufficient_evidence?
+      raise "Method `sufficient_evidence?` should be overridden by the subclass #{self.class}"
+    end
+
+    # Drop-in replacement for `assess_data` using new VRO service
+    def assess_data_with_vro
+      response = vro_client.assess_claim(veteran_icn: claim_context.user_icn)
+      claim_context.assessed_data = response.dig('body', 'evidence').transform_keys(&:to_sym)
+      claim_context.sufficient_evidence = sufficient_evidence?
+    end
+
     # @claim_context.assessed_data has results from assess_data
     def generate_pdf
       # This should call a general PDF generator so that subclasses don't need to override this
       raise "Method `generate_pdf` should be overriden by the subclass #{self.class}"
     end
 
+    # Drop-in replacement for `generate_pdf.render` using new VRO service
+    def generate_pdf_body_with_vro
+      vro_client.generate_summary(veteran_info: claim_context.patient_info, evidence: claim_context.assessed_data)
+      vro_client.download_summary.body
```

Comment:
Will there ever be a case where we call `download_summary` too quickly after the `generate_summary` call?
Yang asked that [here](https://dsva.slack.com/archives/C01Q7979Z7D/p1663179162525329?thread_ts=1663179049.086309&cid=C01Q7979Z7D). 
Response:
> pls wait for the response for the valid response from the POST call before initiating the GET

I asked a follow-on question:
> Is an HTTP response only created after VRO generates the PDF? If so, why is the status "IN_PROGRESS"?
> If not, then let's say VRO is still generating (a big) PDF for a POST call. RRD receives the "IN_PROGRESS" and issues the GET call. Since the PDF is still being generated, what behavior or response should RRD expect?

Reply:
Looks like the response from the VRO side is that there's no situation where the PDF download is issued too soon. Considering this topic resolved here.